.github/workflows/install-valkey/action.yml

```diff
@@ -19,30 +19,52 @@ inputs:
 
 env:
     CARGO_TERM_COLOR: always
-    VALKEY_MIN_VERSION: "7.2.5"
+    VALKEY_MIN_VERSION: "7.2"
 
 runs:
     using: "composite"
 
     steps:
         - name:
               Cache Valkey
-              # TODO: remove the musl ARM64 limitation when https://github.com/actions/runner/issues/801 is resolved
-          if: ${{ inputs.target != 'aarch64-unknown-linux-musl' }}
+              # TODO: self-hosted runners are actually cloning the repo, using the cache from the previous run
+              # will not work as expected. We need to find a way to cache the valkey repo on the runner itself.
+          if: ${{!contains(inputs.target, 'aarch64') }}
           uses: actions/cache@v4
           id: cache-valkey
           with:
               path: |
                   ~/valkey
               key: valkey-${{ inputs.engine-version }}-${{ inputs.target }}
 
+        - name: Build Valkey for ARM
+          if: ${{ contains(inputs.target, 'aarch64') }}
+          shell: bash
+          working-directory: ~
+          run: |
+              cd ~
+              echo "Building valkey ${{ inputs.engine-version }}"
+              # check if the valkey repo is already cloned
+              if [[ ! -d valkey ]]; then
+                  git clone https://github.com/valkey-io/valkey.git
+              fi
+              cd valkey
+              make clean
```

Comment:
run `make clean` inside `if`.

Reply:
Make clean just clean the repo generated files, not the installed engine or similar. Its a good practice and doesn't costs.

Reply:
The cost is complete re-compilation of valkey even if we don't change the version.
When we'll use the cache properly, we'll never have such situation.